cmd/otel-allocator/server/server.go

```diff
@@ -78,6 +79,7 @@ func NewServer(log logr.Logger, allocator allocation.Allocator, discoveryManager
 	router.GET("/jobs", s.JobHandler)
 	router.GET("/jobs/:job_id/targets", s.TargetsHandler)
 	router.GET("/metrics", gin.WrapH(promhttp.Handler()))
+	registerPprof(router.Group("/debug/pprof/"))
```

Comment:
Would it make sense to design the registration in a way that the end point can be switched on and off?

Reply:
is there any reason for it? E.g. perf penalty?

Reply:
As far as I'm aware, just the import has no impact on performance.
When I do see flags for controlling it, it's usually because it's run on a separate port, as the main handler is exposed to untrusted clients or has auth requirements.
The allocator runs with trusted clients, metrics are on the same listener, leading me to the conclusion that there's no need to hide it behind a flag.